### PR TITLE
DPL: close (almost) all file descriptors before forking

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
@chiarazampolli I think this is it.

File: Framework/Core/src/runDataProcessing.cxx [hunk 1]
```diff
@@ -726,20 +726,29 @@ void spawnDevice(DeviceRef ref,
...
+      }
+      if (childFds[ref.index].childstderr[1] == i) {
+        continue;
+      }
+      if (i == STDERR_FILENO) {
+        continue;
+      }
+      auto res = close(i);
+      if (res == 0) {
+        std::cerr << "pid: " << getpid() << "successfully closed " << i << std::endl;
```
Comment (davidrohr) [inline Framework/Core/src/runDataProcessing.cxx]: do we want this printout, or is it just a test?

Comment (ktf) [reply Framework/Core/src/runDataProcessing.cxx]: This was just a test. I will drop it.

**Comment by ktf** [comment]
Tested also on hyperloop. Merging.

**Pull request merged**